sys/kern/link_elf.c

```diff
@@ -446,7 +460,18 @@ link_elf_init(void* arg)
 	/* Compute relative displacement */
 	ef->address = (caddr_t) (__startkernel - KERNBASE);
 #else
+#ifndef __CHERI_PURE_CAPABILITY__
 	ef->address = 0;
+#else /* __CHERI_PURE_CAPABILITY__ */
+	/*
+	 * It is particularly sad that we are not able to put bounds on
+	 * this capability.  At some point ef->address should become a
+	 * plain address and ef should grow capabilities for different
+	 * loadable segments instead.
+	 */
+	ef->address = cheri_setaddress(kernel_root_cap, 0);
+	linker_kernel_file->address = ef->address;
```

Comment:
This line feels like it belongs with the linker_kernel_file lines below

Reply:
Hmm, quite possibly.  It would have to be in the #else again.

Reply:
Moved, should be more readable now.